### PR TITLE
doc(1082): [small] update Branch filtering

### PR DESCRIPTION
**Pull request by tk3fftk**
## Context
SCM Branch-specific design is updated based on https://github.com/screwdriver-cd/screwdriver/issues/1082

## Objective
- Add `~pr:` pattern
- Rename section name to `Branch filtering` from `SCM Branch-specific jobs` with [this discussion](https://github.com/screwdriver-cd/guide/pull/206#discussion_r197592313)
- Add description about workflow configuration in Branch filtering section

## References
- Issue: https://github.com/screwdriver-cd/screwdriver/issues/1082
- Related PR: https://github.com/screwdriver-cd/guide/pull/206

### CODE REVIEW

File: design/workflow.md [hunk 1]
```diff
@@ -52,16 +53,18 @@ jobs:
     requires: ~pr
 ```
 
-### SCM Branch-specific jobs
+### Branch filtering (SCM Branch-specific jobs)
 
-Use exact match of a string (eg: `staging`) or a regex filter (eg: `/^user-.*$/`) after `~commit:` to denote a job only run when code changes are made to branches matching that pattern.
+Use exact match of a string (eg: `staging`) or a regex filter (eg: `/^user-.*$/`) after `~commit:` or `~pr:` to denote a job only run when code changes are made to branches matching that pattern. A workflow configuration is obtained from past sha of a pipeline repositroy if a pipeline restarts from past events which are triggered by branch filtering workflow.
```
Comment (tkyi) [inline design/workflow.md]: pipeline repository*

Comment (Filbird) [inline design/workflow.md]: a past sha*

Please correct me if I'm misunderstanding, but I think the second part of the sentence can be reworded as:

`if an event which was was triggered by a branch filtering workflow is restarted`

Comment (tk3fftk) [reply design/workflow.md]: It's reasonable! fixed!

**Comment by tk3fftk** [comment]
@tkyi fixed typo!

**Pull request merged**